server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -486,31 +493,28 @@ public IndexRequest versionType(VersionType versionType) {
         return this;
     }
 
-    public IndexRequest ifMatch(long seqNo, long term) {
-        if (term == 0 && seqNo != SequenceNumbers.UNASSIGNED_SEQ_NO) {
-            throw new IllegalArgumentException("seqNo is set, but primary term is [0]");
-        }
-
-        if (term != 0 && seqNo == SequenceNumbers.UNASSIGNED_SEQ_NO) {
-            throw new IllegalArgumentException("seqNo is unassigned, but primary term is [" + term + "]");
-        }
+    public IndexRequest setIfSeqNo(long seqNo) {
         if (seqNo < 0 && seqNo != SequenceNumbers.UNASSIGNED_SEQ_NO) {
             throw new IllegalArgumentException("sequence numbers must be non negative. got [" +  seqNo + "].");
         }
+        ifSeqNo = seqNo;
+        return this;
+    }
+
+    public IndexRequest setIfPrimaryTerm(long term) {
         if (term < 0) {
             throw new IllegalArgumentException("primary term must be non negative. got [" + term + "]");
         }
-        ifSeqNoMatch = seqNo;
-        ifPrimaryTermMatch = term;
+        ifPrimaryTerm = term;
         return this;
     }
 
-    public long ifSeqNoMatch() {
-        return ifSeqNoMatch;
+    public long ifSeqNo() {
```

Comment:
can you add javadocs to the new getters and setters in IndexRequest and DeleteRequest?